ambry-server/src/main/java/com.github.ambry.server/AmbryRequests.java

```diff
@@ -671,11 +745,17 @@ private ServerErrorCode validateRequest(PartitionId partition, boolean checkPart
         }
       }
     }
-    // 3. ensure if the partition can be written to
-    if (checkPartitionState && partition.getPartitionState() == PartitionState.READ_ONLY) {
+    // 4. ensure if the partition can be written to
+    if (requestType.equals(RequestOrResponseType.PutRequest)
+        && partition.getPartitionState() == PartitionState.READ_ONLY) {
       metrics.partitionReadOnlyError.inc();
       return ServerErrorCode.Partition_ReadOnly;
     }
+    // 5. Ensure that the request is enabled.
+    if (!isRequestEnabled(requestType, partition)) {
+      metrics.temporarilyUnavailableError.inc();
+      return ServerErrorCode.Temporarily_Unavailable;
```

Comment:
Up to you, just saying that to me `Unavailable` sound not that different from "Unknown" error. It does not provide any insight about the cause of the error. Why reuse error codes? The more distinct we make them, the better.